core/templates/pages/creator-dashboard-page/modal-templates/create-activity-modal.component.spec.ts

```diff
@@ -0,0 +1,137 @@
+import { CreateActivityModalComponent } from './create-activity-modal.component';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { Pipe } from '@angular/core';
+import { UserService } from 'services/user.service';
+import { UserInfo } from 'domain/user/user-info.model';
+
+@Pipe({name: 'translate'})
+class MockTranslatePipe {
+  transform(value: string): string {
+    return value;
+  }
+}
+
+class MockActiveModal {
+  dismiss(): void {
+    return;
+  }
+}
+
+class MockExplorationCreationService {
+  createNewExploration(): void {
+    return null;
+  }
+}
+
+class MockCollectionCreationService {
+  createNewCollection(): void {
+    return null;
+  }
+}
+
+describe('Create Activity Modal Component', () =>{
+  let component: CreateActivityModalComponent;
+  let fixture: ComponentFixture<CreateActivityModalComponent>;
+  let collectionCreationService: CollectionCreationService;
+  let explorationCreationService: ExplorationCreationService;
+  let ngbActiveModal: NgbActiveModal;
+  let userService: UserService;
+
+  beforeEach(fakeAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      declarations: [CreateActivityModalComponent, MockTranslatePipe],
+      providers: [
+        {
+          provide: NgbActiveModal,
+          useClass: MockActiveModal },
+        {
+          provide: ExplorationCreationService,
+          useClass: MockExplorationCreationService
+        },
+        { provide: CollectionCreationService,
+          useClass: MockCollectionCreationService
+        }
+      ]
+    }).compileComponents().then(() => {
+      fixture = TestBed.createComponent(
+        CreateActivityModalComponent);
+      component = fixture.componentInstance;
+    });
+    ngbActiveModal = TestBed.get(NgbActiveModal);
+    explorationCreationService = TestBed.get(ExplorationCreationService);
+    collectionCreationService = TestBed.get(CollectionCreationService);
+    userService = TestBed.get(UserService);
+  }));
+
+  it('should check if user is logged in or not', fakeAsync(() => {
+    const UserInfoObject = {
+      is_moderator: false,
+      is_admin: false,
+      is_super_admin: false,
+      is_topic_manager: false,
+      can_create_collections: true,
+      preferred_site_language_code: null,
+      username: 'tester',
+      email: 'test@test.com',
+      user_is_logged_in: true
+    };
+    spyOn(userService, 'getUserInfoAsync').and.returnValue(Promise.resolve(
+      UserInfo.createFromBackendDict(UserInfoObject))
+    );
+    component.ngOnInit();
+    component.canCreateCollections = true;
+    expect(component.canCreateCollections).toBeTrue();
+  }));
+
+  it('should evalute component properties after component is initialized',
+    () => {
+      component.canCreateCollections = true;
+      expect(component.canCreateCollections).toBeTrue();
```

Comment:
This seems a bit awkward... Why is it required?

Reply:
Fixed